### PR TITLE
Migrate from CircleCI to GHA

### PR DESCRIPTION
**Pull request by cbobach**
Fixes https://github.com/clearhaus/issues-pci/issues/3108

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Deploy
+on:
+  - push
+jobs:
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: eu-west-1
```
Comment (Lassejoe) [inline .github/workflows/deploy.yml:9]: Don't we need `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` as environment variables?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -26,13 +26,19 @@ jobs:
 
       - name: Deploy to AWS
         if: ${{ github.ref == 'refs/heads/master' }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (Lassejoe) [inline .github/workflows/deploy.yml:31]: Just include it after line 10, then you don't have to have it multiple places.

**Comment by cbobach** [comment]
When this is approved I'll disable the CircleCI pipeline.

**Review by Lassejoe — COMMENTED** [review]
based on the dry-run.
I can't really figure out why it needs to upload all these files again, you have only changed the deployment method and nothing else.
So I think something is wrong. I might be wrong.
https://github.com/clearhaus/3DSv2-api-documentation/actions/runs/3143807989/jobs/5108997843

**Comment by cbobach** [comment]
@Lassejoe not sure what you are thinking off here. The dry run is only run on feature branches while the not dry-run one runs only on master.

The old file looks like this
https://github.com/clearhaus/3DSv2-api-documentation/blob/b6ef11fbaa34b854bf60a0bf2629c4c207056631/.circleci/config.yml#L31-L40

This is what I've tried to carry on.

**Comment by Lassejoe** [comment]
> @Lassejoe not sure what you are thinking off here. The dry run is only run on feature branches while the not dry-run one runs only on master.
> 
> The old file looks like this
> 
> https://github.com/clearhaus/3DSv2-api-documentation/blob/b6ef11fbaa34b854bf60a0bf2629c4c207056631/.circleci/config.yml#L31-L40
> 
> This is what I've tried to carry on.

My point is.
The output of the dry-run is questionable.
example:
```
 (dryrun) upload: build/_sources/specification_220.rst.txt to s3://clrhs-prod-nonpci-docs-3dsecure-io/3dsv2/_sources/specification_220.rst.txt
(dryrun) upload: build/_sources/specification_common.rst.txt to s3://clrhs-prod-nonpci-docs-3dsecure-io/3dsv2/_sources/specification_common.rst.txt
```
Since you haven't modified the files the files shouldn't appear in the dry runs output.
It should have been skipped, IMO.
https://docs.aws.amazon.com/cli/latest/reference/s3/sync.html#examples

**Comment by cbobach** [comment]
All the files in `./build/*` is newly generated by the build step and may differ in size, and also timestamp and will thereby be uploaded.

**Review by Lassejoe — APPROVED** [review]

**Comment by cbobach** [comment]
@Lassejoe your thoughts on moving on. Should I just merge this to se if we get something new in the s3 bucket?

I'll remove the unused secret when this is merged.

**Comment by Lassejoe** [comment]
It's just docs. Worst case we revert the branch.
So just go ahead and merge

**Pull request merged**